### ISSUE TITLE
Weeks / Months Support

### ISSUE THREAD

**Issue by flareso** [iamvivekkaushik/DatePickerTimelineFlutter#43]
Is it possible to pick type of the time?

for example. type.daily / type.weekly / type.monthly

**Comment by dvird** [comment]
@iamvivekkaushik is this project still maintained by you?

**Comment by iamvivekkaushik** [comment]
> @iamvivekkaushik is this project still maintained by you?

Yes, In my free time I still work on it. I don't understand what you mean by "pick type of time".

**Comment by dvird** [comment]
Great !
See here some did pull request for month selection.
im mean we'll being able to selection the presentation type like calendar. daily selection (which is what we've right now), weekly selection (weeks scroll view), and months

**Comment by iamvivekkaushik** [comment]
No, it does not support that.

**Comment by iamvivekkaushik** [comment]
Quick update on this old thread: the project is very much alive — v1.3.0 just shipped with a bunch of fixes (tile overflow, controller crashes, `didUpdateWidget`) and a new showPastDates option that renders past dates and centers the selection.

On the actual request: the picker is still strictly day-based — one tile per day, no weekly or monthly presentation modes. The calendarType parameter only switches between Gregorian and Persian calendars, not the granularity. I'm keeping this open as the tracking issue for it.

**Comment by iamvivekkaushik** [comment]
This finally exists — **v1.4.0** adds a `granularity` parameter with exactly
the three presentation types you asked for:

```dart
DatePicker(
  DateTime.now(),
  granularity: DateGranularity.week,  // .day (default) / .week / .month
  firstDayOfWeek: DateTime.monday,    // optional; defaults to the locale's
  daysCount: 26,                      // now counts tiles: weeks here
  onDateChange: (weekStart) => ...,   // always the unit's first day
)
```
